src/main/java/org/folio/consortia/service/impl/TenantManagerImpl.java

```diff
@@ -102,17 +103,29 @@ public Tenant update(UUID consortiumId, String tenantId, Tenant tenantDto) {
 
   @Override
   @Transactional
-  public void delete(UUID consortiumId, String tenantId) {
+  public void delete(UUID consortiumId, String tenantId, TenantDeleteRequest tenantDeleteRequest) {
     consortiumService.checkConsortiumExistsOrThrow(consortiumId);
     var tenant = getTenantById(tenantId);
-    validateTenantForDeleteOperation(tenant);
+    var isHardDelete = tenantDeleteRequest.getDeleteType().equals(DeleteTypeEnum.HARD);
```

Comment:
let's make tenantDeleteRequest.getDeleteType() as required param  and use switch, probably it can simplify a bit reading bollean conditions with flags